test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java

```diff
@@ -133,6 +152,61 @@ public static void teardown() {
     SharedMiniClusterBase.stopMiniCluster();
   }
 
+  @Test
+  public void testEventualScan() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String table = getUniqueNames(1)[0];
+      client.tableOperations().create(table);
+
+      // Start Scan Server
+      getCluster().getClusterControl().start(ServerType.SCAN_SERVER);
+
+      // generate 100 rows of data
+      write(client, table, generateMutations(0, 100, tr -> true));
+
+      // need flush table so that eventual scan can see it.
+
+      long minWait = 5_000;
+      var startTime = System.currentTimeMillis();
+      client.tableOperations().flush(table, null, null, true);
+      var duration = System.currentTimeMillis() - startTime;
+      // if duration is less than minWait, wait the remainder + 0.5 seconds.
+      if (minWait > duration) {
+        Thread.sleep((minWait - duration) + 500);
+      }
+
+      // should see all data that was flushed in eventual scan
+      verifyData(client, table, AUTHORIZATIONS, generateKeys(0, 100),
+          scanner -> scanner.setConsistencyLevel(EVENTUAL));
+
+      // should not see data with col vis set
+      verifyData(client, table, Authorizations.EMPTY, generateKeys(0, 100, tr -> tr.vis.isEmpty()),
+          scanner -> scanner.setConsistencyLevel(EVENTUAL));
+
+      // write some more rows after 100 and verfy those are not seen by eventual scan until table is
+      // flushed.
+      write(client, table, generateMutations(100, 200, tr -> true));
+      verifyData(client, table, AUTHORIZATIONS, generateKeys(0, 100),
+          scanner -> scanner.setConsistencyLevel(EVENTUAL));
+
+      // need flush table so that eventual scan can see it.
+      startTime = System.currentTimeMillis();
+      client.tableOperations().flush(table, null, null, true);
+      duration = System.currentTimeMillis() - startTime;
+      // if duration is less than minWait, wait the remainder + 0.5 seconds.
+      if (minWait > duration) {
+        Thread.sleep((minWait - duration) + 500);
+      }
```

Comment:
Instead of sleeping after the flush could instead wait for the data to be present like the following.

```suggestion
// wait for the eventual scan to see the new data
      final int initialSize = generateKeys(0, 100).size();
      Wait.waitFor(()-> {
        try(var scanner = client.createScanner(table, AUTHORIZATIONS)) {
          scanner.setConsistencyLevel(EVENTUAL);
          return scan(scanner).size() > initialSize;
        }
      });
```